src/libraries/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -279,6 +301,48 @@ private static void OnRequestSendingRequest(WinHttpRequestState state)
                 var serverCertificate = new X509Certificate2(certHandle);
                 Interop.Crypt32.CertFreeCertificateContext(certHandle);
 
+                IPAddress? ipAddress = null;
+                unsafe
+                {
+                    Interop.WinHttp.WINHTTP_CONNECTION_INFO connectionInfo;
+                    Interop.WinHttp.WINHTTP_CONNECTION_INFO* pConnectionInfo = &connectionInfo;
+                    uint infoSize = (uint)sizeof(Interop.WinHttp.WINHTTP_CONNECTION_INFO);
+                    if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(state, $"sizeof(WINHTTP_CONNECTION_INFO)={infoSize}");
+                    if (Interop.WinHttp.WinHttpQueryOption(
+                        state.RequestHandle,
+                        Interop.WinHttp.WINHTTP_OPTION_CONNECTION_INFO,
+                        (IntPtr)pConnectionInfo,
+                        ref infoSize))
+                    {
+                        ReadOnlySpan<byte> RemoteAddressSpan = new ReadOnlySpan<byte>(connectionInfo.RemoteAddress, 128);
+                        AddressFamily addressFamily = (AddressFamily)BitConverter.ToInt16(RemoteAddressSpan.ToArray(), 0);
```

Comment:
We should avoid the `ToArray()` allocation. Shouldn't we deal with endiannes here too and use `BinaryPrimitives`?